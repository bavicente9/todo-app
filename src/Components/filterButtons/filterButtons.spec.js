import componentWithMockStore from "../../Utils/test/componentWithMockStore";
import renderer from 'react-test-renderer';
import FilterButtons from "./filterButtons";




describe('filter buttons', () => {
    let store
    let component;
    const simulateClickByProps = (props) => {
        //example Of props { onclick: handleShowActive() }
        renderer.act(
            component.root.findByProps().props.onclick()
        )
    }
    
    beforeEach(() => {
        //render a component with a mock Store
        ({ component, store } = componentWithMockStore(<FilterButtons />))
    });

    it('match with snapshot', () => {
        expect(component.toJSON()).toMatchSnapshot();
    })


    it('correctly call of All button dispatch', () => {

        renderer.act(
            component.root.findByProps({ onclick: handleShowCompleted() }).props.onclick()
        )
        expect(store.dispatch).toHaveBeenCalledTimes(1);
    })

    it('correctly call of completed button dispatch', () => {
        renderer.act(
            component.root.findByProps({ onclick: handleShowCompleted() }).props.onclick()
        )
        expect(store.dispatch).toHaveBeenCalledTimes(1);

    })
    
    
    it('correctly call of Active button dispatch', () => {
        //TODO: check if this fn work, if it work refactoring the others tests
        simulateClickByProps({onclick:handleShowActive()});

        expect(store.dispatch).toHaveBeenCalledTimes(1);
    })


    
})