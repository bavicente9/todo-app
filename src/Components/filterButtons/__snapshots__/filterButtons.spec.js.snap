// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filter buttons match with snapshot 1`] = `
<div
  className="filterButtons_container filterButtons_container--dark undefined"
>
  <button
    className="filter_button "
    onClick={[Function]}
    value="all"
  >
    All
  </button>
  <button
    className="filter_button  "
    onClick={[Function]}
    value="active"
  >
    Active
  </button>
  <button
    className="filter_button "
    onClick={[Function]}
    value="completed"
  >
    Completed
  </button>
</div>
`;
